components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/add-security-questions.jsp

```diff
@@ -0,0 +1,157 @@
+<%--
+  ~ Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+  ~
+  ~ WSO2 Inc. licenses this file to you under the Apache License,
+  ~ Version 2.0 (the "License"); you may not use this file except
+  ~ in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  --%>
+<%@ page contentType="text/html;charset=UTF-8" pageEncoding="UTF-8" %>
+
+<%@ page import="org.owasp.encoder.Encode" %>
+
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.AuthenticationEndpointUtil" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.EncodedControl" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.Constants" %>
+<%@ page import="org.wso2.carbon.identity.recovery.ui.IdentityManagementAdminClient" %>
+<%@ page import="org.wso2.carbon.identity.recovery.model.ChallengeQuestion" %>
+<%@ page import="org.wso2.carbon.utils.ServerConstants" %>
+<%@ page import="org.wso2.carbon.ui.CarbonUIUtil" %>
+<%@ page import="org.wso2.carbon.CarbonConstants" %>
+
+<%@ page import="java.util.ResourceBundle" %>
+<%@ page import="java.nio.charset.StandardCharsets" %>
+
+<%@ page import="java.util.ArrayList" %>
+<%@ page import="java.util.Iterator" %>
+<%@ page import="java.util.HashMap" %>
+<%@ page import="java.util.List" %>
+<%@ page import="java.util.Map" %>
+
+<%
+    String BUNDLE = "org.wso2.carbon.identity.application.authentication.endpoint.i18n.Resources";
+    ResourceBundle resourceBundle = ResourceBundle.getBundle(BUNDLE, request.getLocale(), new
+            EncodedControl(StandardCharsets.UTF_8.toString()));
+    String urlData = request.getParameter("data");
+    // Extract the challenge questions from the request and add them into an array
+    String[] questionSets = urlData.split("&");
+    // Hash-map to hold user's challenge questions
+    Map<String, List<ChallengeQuestion>> challengeQuestionMap = new HashMap<>();
+
+    for (String question : questionSets) {
+        String[] questionProperties = question.split("\\|");
+        // Construct a new ChallengeQuestion for each challenge question received from the request
+        ChallengeQuestion tempChallengeQuestions = new ChallengeQuestion();
+        tempChallengeQuestions.setQuestionSetId(questionProperties[0]);
```

Comment:
If we can extract a variable with questionProperties[0] having name questionSetId it would make the logic more readable

Reply:
fixed